src/Segmentation/regionprops.jl

```diff
@@ -297,3 +311,413 @@ function converttounits!(propdf, latlondata, colstodrop)
+location of the label.
+
+Examples:
+```jldoctest; setup = :(using IceFloeTracker)
+
+julia> A = zeros(Int, 8, 6); A[2:6, 1:2] .= 1; A[3:7, 4:5] .= 2; A[4:6, 3:6] .= 2;
+julia> A
+8×6 Matrix{Int64}:
+ 0  0  0  0  0  0
+ 1  1  0  0  0  0
+ 1  1  0  2  2  0
+ 1  1  2  2  2  2
+ 1  1  2  2  2  2
+ 1  1  2  2  2  2
+ 0  0  0  2  2  0
+ 0  0  0  0  0  0
+
+julia> masks = component_floes(A)
+julia> Dict{Any, Any} with 3 entries:
+  0 => Bool[1 1 … 1 1; 0 0 … 1 1; … ; 1 1 … 0 1; 1 1 … 1 1]
+  2 => Bool[0 1 1 0; 1 1 1 1; … ; 1 1 1 1; 0 1 1 0]
+  1 => Bool[1 1; 1 1; … ; 1 1; 1 1]
+julia> masks[2]
+5×4 BitMatrix:
+ 0  1  1  0
+ 1  1  1  1
+ 1  1  1  1
+ 1  1  1  1
+ 0  1  1  0
+julia> masks[1]
+5×2 BitMatrix:
+ 1  1
+ 1  1
+ 1  1
+ 1  1
+ 1  1
+```
+"""
+function component_floes(indexmap; minimum_area=1)
+    labels = unique(indexmap)
+    boxes = component_boxes(indexmap)
+    mn, mx = extrema(indexmap)
+    if !(mn == 0 || mn == 1)
+        throw(ArgumentError("The input labeled array should contain background label `0` as the minimum value"))
+    end
+    areas = component_lengths(indexmap)
+    floe_masks = Dict()
+    for i in labels
+        if areas[i] > minimum_area
+            floe_masks[i] = indexmap[boxes[i]] .== i
+        end
+    end
+    return floe_masks
+end
+
+"""component_perimeter(indexmap; minimum_area=1, algorithm="benkrid_crookes")
+
+Estimate the perimeter of the labeled regions in `indexmap` using the specified algorithm.
+Algorithm options = "benkrid_crookes" (only option currently, will add crofton in future release)
+
+
+"""
+function component_perimeters(
+    indexmap;
+    minimum_area=1, 
+    algorithm="benkrid_crookes",
+    connectivity=4)
+    masks = component_floes(indexmap)
+    perims = Dict()
+    algorithm ∉ ["benkrid_crookes"] && begin
+        print("Unsupported Algorithm, defaulting to Benkrid-Crookes")
+        algorithm = "benkrid_crookes"
+    end
+
+    algorithm_functions = Dict("benkrid_crookes" => benkrid_crookes)
+
+    for label in keys(masks)
+        n, m = size(masks[label])
+        n * m == 1 && continue
+        label == 0 && (masks[label] = 0; continue)
+        
+        # Shape needs to have a border of zeros for erode to work here    
+        mpad = padarray(masks[label], Fill(0, (1, 1)))
+        connectivity == 4 ? (strel = strel_diamond((3,3))) : (strel = strel_box((3,3)))
+        epad = mpad .- erode(mpad, strel)
+        e = epad[1:n, 1:m]
+        perims[label] = algorithm_functions[algorithm](e)
+    end
+    return perims
+end
+
+function benkrid_crookes(edge_array)
+    type_vals = zeros(33)
+    type_vals[[5, 7, 15, 17, 25, 27]] .= 1
+    type_vals[[21, 33]] .= sqrt(2)
+    type_vals[[13, 23]] .= (1 + sqrt(2)) / 2
+
+    conv_arr = [10 2 10; 2 1 2; 10 2 10]
+    results = conv(edge_array, conv_arr; algorithm=:direct)
+
+    val_counts = Dict()
+    for val in vec(results)
+        val ∉ keys(val_counts) && (val_counts[val] = 0)
+        val_counts[val] += 1
```

Comment:
Here one could use the `get` function with default value, akin to the `.get` method for Python dicts.
```suggestion
        val_counts[val] = get(val_counts, val, 0) + 1
```